pennylane/devices/qubit/apply_operation.py

```diff
@@ -198,6 +199,31 @@ def _(op: type_op, state):
     return _apply_operation_default(op, state, is_state_batched, debugger)
 
 
+@apply_operation.register
+def apply_mid_measure(op: MidMeasureMP, state, is_state_batched: bool = False, debugger=None):
+    """Applies a native mid-circuit measurement."""
+    from .measure import measure  # pylint: disable=import-outside-toplevel
+
+    if is_state_batched:
+        raise ValueError("MidMeasureMP cannot be applied to batched states.")
+    wire = op.wires
+    probs = measure(qml.probs(wire), state)
+    dark_branch = np.random.binomial(1, probs[0])
```

Comment:
It makes sense why you're using the dark branch for the logic, but in terms of readability it might make more sense to either add a comment here or use the "light" branch.

Reply:
I flipped the bits and simply called it `sample`. Is that better? Or would we prefer `chosen_branch` or something else?